llama_index/agent/runner/base.py

```diff
@@ -208,15 +208,33 @@ def __init__(
         init_task_state_kwargs: Optional[dict] = None,
         delete_task_on_finish: bool = False,
         default_tool_choice: str = "auto",
+        verbose: bool = False,
     ) -> None:
         """Initialize."""
         self.agent_worker = agent_worker
         self.state = state or AgentState()
         self.memory = memory or ChatMemoryBuffer.from_defaults(chat_history, llm=llm)
-        self.callback_manager = callback_manager or CallbackManager([])
+
+        # get and set callback manager
+        if callback_manager is not None:
```

Comment:
we should reeealllly just make the callback manager global. This is possible with v0.10.0

Theres a ton of brittle callback manager inheritance going on lol

Reply:
true. will take a deeper look when i make the agents video